server/src/main/java/org/opensearch/extensions/rest/RestSendToExtensionAction.java

```diff
@@ -259,7 +251,12 @@ public String executor() {
                     filteredHeaders,
                     contentType,
                     content,
-                    requestIssuerIdentity,
+                    identityService.getTokenManager()
+                        .issueOnBehalfOfToken(Map.of(StandardTokenClaims.AUDIENCE.getName(), discoveryExtensionNode.getId())) // This gets
+                                                                                                                              // an
+                                                                                                                              // extensions
+                                                                                                                              // uniqueId
+                        .toString(),
```

Comment:
`toString()` is not the right method to call here. This is the `toString()` on BearerAuthToken currently: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/identity/tokens/BearerAuthToken.java#L57-L60. There may need to be a method on `AuthToken` to get the value of the token and not its `toString()` representation.

IMO this should pass an AuthToken that is writeable and can be understood by the SDK via NamedWriteableRegistry, but I think it can also be assumed that this is a JWT (Bearer) token if we don't want to add in the AuthToken abstraction initially.

Reply:
Any thoughts on this: https://github.com/scrawfor99/OpenSearch/pull/6?